config/initializers/paperclip.rb

```diff
@@ -40,6 +42,8 @@
   s3_protocol = ENV.fetch('S3_PROTOCOL') { 'https' }
   s3_hostname = ENV.fetch('S3_HOSTNAME') { "s3-#{s3_region}.amazonaws.com" }
 
+  Paperclip::Attachment.default_options[:path] = ENV.fetch('S3_KEY_PREFIX') + "/#{PATH}" if ENV.has_key?('S3_KEY_PREFIX')
```

Comment:
Instead of the explicit setting of this option - should this just modify the existing interpolation for `prefix_url`?

Separately - this line could use a default nil/blank value in the fetch, instead of checking if var is set?

Reply:
> Other places in this file use the if ENV.has_key? pattern.

Fair point, I hadn't looked at full file. Agree that keeping style consistent makes sense.

> having it set in the S3 block also reduces possible side effects to a minimum.

Also good point.